backend/pkg/logger/messages/logger.go

```diff
@@ -0,0 +1,113 @@
+package messages
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/logger"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/packet/info"
+)
+
+const (
+	Name abstraction.LoggerName = "messages"
+)
+
+type Logger struct {
+	// An atomic boolean is used in order to use CompareAndSwap in the Start and Stop methods
+	running  *atomic.Bool
+	fileLock *sync.RWMutex
+	// initialTime fixes the starting time of the log
+	initialTime time.Time
+	// infoIdMap is a map that contains the file of each info packet
```

Comment:
Maybe this comment is better
```suggestion
	// infoIdMap is a map that maps each info packet with a file
```